### PR TITLE
Fix for buildMultiTargeting

### PR DESCRIPTION
**Pull request by Mikejo5000**

### CODE REVIEW

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[cdcc388](https://github.com/NuGet/docs.microsoft.com-nuget/commits/cdcc3887648d9d2dcda2865caa009172ed82bf63)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C9%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908092150072750-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Nq%2By34b3Oul4zUxfz4sT%2FZxTdlv0JAv6wBtAfj6ruPY%3D&st=2019-08-09T21%3A46%3A04Z&se=2019-09-09T21%3A51%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by Mikejo5000** [comment]
Related issues:
#1508 and #777 and #231

**Comment by Mikejo5000** [comment]
@nkolev92 There are several requests for updates around this. What's unclear to me is whether we need to say anything more about MSBuild props and targets in this topic:
https://docs.microsoft.com/en-us/nuget/create-packages/multiple-target-frameworks-project-file
Step 4. mentions .props and .targets. I'm not sure if the content here is sufficient. It appears to me that the "buildMultiTargeting" folder is only relevant if using a .nuspec.

**Comment by casz** [comment]
@Mikejo5000 I think @natemcmaster explains the need for `buildMultiTargeting` folder just fine: https://natemcmaster.com/blog/2017/07/05/msbuild-task-in-nuget/

**Comment by nkolev92** [comment]
It's relevant in both csproj(msbuild) pack and nuspec pack cases. 

The need for buildMultiTargetting is driven by restore side, not the pack method. 
Similar thing applies for buildTransitive.

**Comment by Mikejo5000** [comment]
@nkolev92 So is the change in this PR helpful? Or is there more that needs to be said about it?

File: docs/create-packages/Creating-a-Package.md [hunk 1]
```diff
@@ -180,6 +180,7 @@ The folder conventions are as follows:
 | runtimes | Architecture-specific assembly (`.dll`), symbol (`.pdb`), and native resource (`.pri`) files | Assemblies are added as references only for runtime; other files are copied into project folders. There should always be a corresponding (TFM) `AnyCPU` specific assembly under `/ref/{tfm}` folder to provide corresponding compile time assembly. See [Supporting multiple target frameworks](supporting-multiple-target-frameworks.md). |
 | content | Arbitrary files | Contents are copied to the project root. Think of the **content** folder as the root of the target application that ultimately consumes the package. To have the package add an image in the application's */images* folder, place it in the package's *content/images* folder. |
 | build | MSBuild `.targets` and `.props` files | Automatically inserted into the project file or `project.lock.json` (NuGet 3.x+). |
+| buildMultiTargeting | MSBuild `.targets` and `.props` files for cross-framework targeting | Automatically inserted into the project file or `project.lock.json` |
```
Comment (nkolev92) [inline docs/create-packages/Creating-a-Package.md]: > project file or `project.lock.json`

We should remove all references to `project.lock.json`. 

It's a leftover from project.json, PackageReference has some differences."

I'd say both build and buildMultiTargeting should say "into the project". 

The file bit is incorrect. We don't really edit the file.

Comment (nkolev92) [reply docs/create-packages/Creating-a-Package.md]: We should add an explanation for buildTransitive as well.

File: docs/create-packages/Creating-a-Package.md [hunk 2]
```diff
@@ -342,6 +343,14 @@ When NuGet installs a package with `\build` files, it adds MSBuild `<Import>` el
...
 
+    \buildMultiTargeting
+        \netstandard1.4
+            \Contoso.Utility.UsefulStuff.props
+            \Contoso.Utility.UsefulStuff.targets
+        \net462
+            \Contoso.Utility.UsefulStuff.props
+            \Contoso.Utility.UsefulStuff.targets
+
 With NuGet 3.x, targets are not added to the project but are instead made available through the `project.lock.json`.
```
Comment (nkolev92) [inline docs/create-packages/Creating-a-Package.md]: > With NuGet 3.x, targets are not added to the project but are instead made available through the `project.lock.json`.

Through {projectName}.nuget.g.targets and {projectName}.nuget.g.props.

File: docs/create-packages/Creating-a-Package.md [hunk 3]
```diff
@@ -342,6 +343,14 @@ When NuGet installs a package with `\build` files, it adds MSBuild `<Import>` el
 
 MSBuild `.props` and `.targets` files for cross-framework targeting can be placed in the `\buildMultiTargeting` folder. During package installation, NuGet adds the corresponding `<Import>` elements to the project file with the condition, that the target framework is not set (the MSBuild property `$(TargetFramework)` must be empty).
 
+    \buildMultiTargeting
```
Comment (nkolev92) [inline docs/create-packages/Creating-a-Package.md]: Not sure why we need to add this necessarily? 
Why not add it with build?

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[79e8c2b](https://github.com/NuGet/docs.microsoft.com-nuget/commits/79e8c2b208dc42a3d257edff41a8ae04026fd96f)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C13%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908131616226715-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=ihnhLexAhqF9yE5vr9yi1ROOeL461gDg6q%2BPVLW4wik%3D&st=2019-08-13T16%3A12%3A23Z&se=2019-09-13T16%3A17%3A23Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[4c74bf2](https://github.com/NuGet/docs.microsoft.com-nuget/commits/4c74bf202e5f257d5460a723c41ef2cbae33fc0b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C13%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908131625141123-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Y2eaZXZqTnPYhriTmJBehHjtOlSJ52FPPQKiW5x50uQ%3D&st=2019-08-13T16%3A21%3A14Z&se=2019-09-13T16%3A26%3A14Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[e5dfc98](https://github.com/NuGet/docs.microsoft.com-nuget/commits/e5dfc982111266cc97d7f0b4d3858033fe3077dd)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/consume-packages/Package-References-in-Project-Files.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/consume-packages/Package-References-in-Project-Files.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files?branch=pr-en-us-1616) |
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C13%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908131635535570-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Cwr7xuQpn%2BWLvXugkj5CcW8yLU7MfiFSzEjvk5n8Zmg%3D&st=2019-08-13T16%3A32%3A05Z&se=2019-09-13T16%3A37%3A05Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by Mikejo5000** [comment]
@nkolev92 Incorporated feedback. Could you take a look at changes for buildTransitive? I'd like to fix #1476 at the same time, if possible.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[262714a](https://github.com/NuGet/docs.microsoft.com-nuget/commits/262714ac625ecd7df924cb0f80d654ea5d8bc0ff)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/consume-packages/Package-References-in-Project-Files.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/consume-packages/Package-References-in-Project-Files.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files?branch=pr-en-us-1616) |
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C13%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908131752354646-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=DODD%2BLl6kK7Go5RilZ%2BIbz%2Bu0SEjQTMsG0Ns6lLw7QI%3D&st=2019-08-13T17%3A48%3A34Z&se=2019-09-13T17%3A53%3A34Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[b634ee1](https://github.com/NuGet/docs.microsoft.com-nuget/commits/b634ee180724acd8a3dde5e336ae064a703f0b50)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/consume-packages/Package-References-in-Project-Files.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/consume-packages/Package-References-in-Project-Files.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files?branch=pr-en-us-1616) |
[docs/create-packages/Creating-a-Package.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/mikejo-multi/docs/create-packages/Creating-a-Package.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/creating-a-package?branch=pr-en-us-1616) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C14%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908141612119838-1616%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=3agUrpXZHhvKBJd0Op7K1cDx4YmI45BBIfOAG0ak3ts%3D&st=2019-08-14T16%3A08%3A08Z&se=2019-09-14T16%3A13%3A08Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/consume-packages/Package-References-in-Project-Files.md [hunk 4]
```diff
@@ -100,7 +100,9 @@ Allowable values for these tags are as follows, with multiple values separated b
 | compile | Contents of the `lib` folder and controls whether your project can compile against the assemblies within the folder |
 | runtime | Contents of the `lib` and `runtimes` folder and controls whether these assemblies will be copied out to the build output directory |
 | contentFiles | Contents of the `contentfiles` folder |
-| build | Props and targets in the `build` folder |
+| build | `.props` and `.targets` in the `build` folder |
+| buildMultitargeting | `.props` and `.targets` in the `buildMultitargeting` folder, for cross-framework targeting |
```
Comment (zivkan) [inline docs/consume-packages/Package-References-in-Project-Files.md]:
```suggestion
| buildMultiTargeting | *(4.0+)* `.props` and `.targets` in the `buildMultiTargeting` folder, for cross-framework targeting |
```

File: docs/create-packages/Creating-a-Package.md [hunk 5]
```diff
@@ -179,7 +179,9 @@ The folder conventions are as follows:
...
 | runtimes | Architecture-specific assembly (`.dll`), symbol (`.pdb`), and native resource (`.pri`) files | Assemblies are added as references only for runtime; other files are copied into project folders. There should always be a corresponding (TFM) `AnyCPU` specific assembly under `/ref/{tfm}` folder to provide corresponding compile time assembly. See [Supporting multiple target frameworks](supporting-multiple-target-frameworks.md). |
 | content | Arbitrary files | Contents are copied to the project root. Think of the **content** folder as the root of the target application that ultimately consumes the package. To have the package add an image in the application's */images* folder, place it in the package's *content/images* folder. |
-| build | MSBuild `.targets` and `.props` files | Automatically inserted into the project file or `project.lock.json` (NuGet 3.x+). |
+| build | MSBuild `.targets` and `.props` files | Automatically inserted into the project (NuGet 3.x+). |
+| buildMultiTargeting | MSBuild `.targets` and `.props` files for cross-framework targeting | Automatically inserted into the project. |
```
Comment (zivkan) [inline docs/create-packages/Creating-a-Package.md]:
```suggestion
| buildMultiTargeting | *(4.0+)* MSBuild `.targets` and `.props` files for cross-framework targeting | Automatically inserted into the project. |
```

**Comment by Mikejo5000** [comment]
Linking to buildTransitive feature page on the Wiki, at least for now, instead of trying to explain it here seems best.

**Pull request merged**